### DISCUSSION TITLE
Cockroachdb support

### DISCUSSION THREAD

**Discussion by rdlaitila** [geldata/gel#3403] [Ideas]
Could not find any mention of Cockroachdb support in discussions/issues, wondering if its on the project's radar. I think EdgeDB + Cockroachdb's scale out model could be a nice pairing. Thanks!

**Comment by elprans** [comment]
Using CockroachDB as the backend for EdgeDB would definitely be an interesting proposition, however this would likely require a lot of work on both sides.

EdgeDB _does not_ treat Postgres as a simple standard SQL store.  The opposite is true.  To realize the full potential of the graph-relational model and EdgeQL efficiently, we must squeeze every last bit of functionality out of PostgreSQL's implementation of SQL and its schema.  Even then we've bumped into the ceiling quite a few times, and needed to send patches and bug reports upstream.  That's also why EdgeDB currently requires PostgreSQL 13 or later.

In other words, EdgeDB and Postgres are like Rust and LLVM: porting to a different backend could only be completely successful if the new backend is at least as capable as the old in the relevant dimensions.

That said, EdgeDB, the implementation, and EdgeQL, the language and the model, are different, and we encourage research into alternative EdgeQL implementations.  We'll be releasing the formal EdgeQL spec and the graph-relational model whitepaper to make this easier.
